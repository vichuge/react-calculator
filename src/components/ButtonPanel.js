import React from 'react';
import Button from './Button';

const ButtonPanel = () => (
  <Button buttonName="test" type="type1" />
);

export default ButtonPanel;
