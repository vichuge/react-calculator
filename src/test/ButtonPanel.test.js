import React from 'react';
import renderer from 'react-test-renderer';
import ButtonPanel from '../components/ButtonPanel';

it('return button panel', () => {
  const handleClick = (buttonName) => {
    console.log(buttonaName);
  };
  const tree = renderer
    .create(<ButtonPanel clickApp={handleClick} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});