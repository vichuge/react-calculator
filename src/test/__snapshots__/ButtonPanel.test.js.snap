// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`return button panel 1`] = `
Array [
  <div
    className="g1"
  >
    <button
      className="gray "
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="gray "
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="gray "
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="orange "
      onClick={[Function]}
      type="button"
    >
      /
    </button>
  </div>,
  <div
    className="g1"
  >
    <button
      className="gray "
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="gray "
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="gray "
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="orange "
      onClick={[Function]}
      type="button"
    >
      X
    </button>
  </div>,
  <div
    className="g1"
  >
    <button
      className="gray "
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="gray "
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="gray "
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="orange "
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>,
  <div
    className="g1"
  >
    <button
      className="gray "
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="gray "
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="gray "
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="orange "
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>,
  <div
    className="g1"
  >
    <button
      className="zero "
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="gray "
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="orange "
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>,
]
`;
